kaldb/src/main/java/com/slack/kaldb/writer/kafka/KaldbKafkaConsumer.java

```diff
@@ -211,9 +212,32 @@ public void consumeMessages() throws IOException {
     consumeMessages(KAFKA_POLL_TIMEOUT_MS);
   }
 
+  protected ConsumerRecords<String, byte[]> pollWithRetry(final long kafkaPollTimeoutMs) {
+    ConsumerRecords<String, byte[]> records = null;
+    OffsetOutOfRangeException kafkaError = null;
+    for (int i = 0; i < 4; i++) {
+      try {
+        records = kafkaConsumer.poll(Duration.ofMillis(kafkaPollTimeoutMs));
+        kafkaError = null;
+        break;
+      } catch (OffsetOutOfRangeException ex) {
+        kafkaError = ex;
+        LOG.warn("Caught unexpected exception, retry number: {}, exception: {}", i, ex);
```

Comment:
small nit: I think something like this will get formatted correct i.e the exception part of it

```
LOG.warn(String.format("Caught unexpected exception, retry number: %i", i), ex);
```

That way the `void warn(String var1, Throwable var2);` signature applies.

Reply:
The suggestion looks good to me, updated the code .